pkg/reconciler/ingress/resources/virtual_service.go

```diff
@@ -33,80 +33,147 @@ import (
 	"github.com/knative/serving/pkg/apis/networking/v1alpha1"
 	"github.com/knative/serving/pkg/apis/serving"
 	"github.com/knative/serving/pkg/network"
-	"github.com/knative/serving/pkg/reconciler/clusteringress/resources/names"
+	"github.com/knative/serving/pkg/reconciler/ingress/resources/names"
 	"github.com/knative/serving/pkg/resources"
 )
 
-// VirtualServiceNamespace gives the namespace of the child
+// ClusterIngressVirtualServiceNamespace gives the namespace of the child
 // VirtualServices for a given ClusterIngress.
-func VirtualServiceNamespace(_ *v1alpha1.ClusterIngress) string {
+func ClusterIngressVirtualServiceNamespace(_ *v1alpha1.ClusterIngress) string {
 	return system.Namespace()
 }
 
-// MakeIngressVirtualService creates Istio VirtualService as network
+// IngressVirtualServiceNamespace gives the namespace of the child
+// VirtualServices for a given Ingress.
+func IngressVirtualServiceNamespace(i *v1alpha1.Ingress) string {
+	return i.Namespace
+}
```

Comment:
This seems kinda redundant, why not use `i.Namespace` on the calling side directly?